### PR TITLE
Unlink build and release workflows from internal distribution process

### PR DESCRIPTION
**Pull request by dpotyralski**
Due to internal changes with GitHub workflows and the missing possibility to utilise reusable workflows from private repositories. Right now it has been decided to unlink the existing workflows from `centralized-templates` and make those currently existing ones repository-specific.

Linear story: [DEVX-2677](https://linear.app/pleo/issue/DEVX-2677)

Have you done ...
* [ ] Unit tests

### CODE REVIEW

File: .github/workflows/build-and-test.yaml [hunk 1]
```diff
@@ -138,24 +138,6 @@ jobs:
...
-          JOB_RUNR_REPO_PASSWORD: ${{ secrets.JOB_RUNR_REPO_PASSWORD }}
-
-      # Publish Detekt results to GitHub.
-      - name: Publish Detekt SARIF report to GitHub
-        continue-on-error: true
-        if: ${{ github.ref == 'refs/heads/main' }}
-        uses: github/codeql-action/upload-sarif@d39d31e687223d841ef683f52467bd88e9b21c14 # v3
-        with:
-          sarif_file: build/reports/detekt/detekt.sarif
-
```
Comment (dpotyralski) [inline .github/workflows/build-and-test.yaml:158]: Those reports are not in use.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -413,291 +409,3 @@ jobs:
           SLACK_USERNAME: GitHub Actions
           SLACK_ICON_EMOJI: ":crashingrocket:"
           SLACK_FOOTER: ""
-
-  publish-docker-image:
```
Comment (dpotyralski) [inline .github/workflows/release.yaml:417]: We don't need to publish a Docker image from the library.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -413,291 +409,3 @@ jobs:
...
-      environmentTenant: ${{ steps.env-variables.outputs.environmentTenant }}
-    runs-on: ubuntu-latest
-    steps:
-      - name: Set output variables
-        id: env-variables
-        run: |
-          echo "deploymentEnabled=${{ env.deploymentEnabled }}" >> "$GITHUB_OUTPUT"
-          echo "environmentTenant=${{ env.environmentTenant }}" >> "$GITHUB_OUTPUT"
-
-  wiz-cli:
```
Comment (dpotyralski) [inline .github/workflows/release.yaml:597]: There is now no image to scan.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -413,291 +409,3 @@ jobs:
...
-    needs:
-      - publish-docker-image
-      - get-env-variables
-    with:
-      scan_container: false
-      container_tag: ${{ github.sha }}
-      deps_policy: Pleo-Default-vulnerabilities-policy
-    secrets: inherit
-
-  trigger-deploy-gitops-product-dev:
```
Comment (dpotyralski) [inline .github/workflows/release.yaml:609]: There is now, no deployment to trigger.

**Review by r0binary — APPROVED** [review]

**Comment by DimitrisKaramanis** [comment]
@dpotyralski Is this a copy-paste from the new cicd workflow, or you manipulated the files that exist here?

I think there are 2 "safe" options. Either keep everything as it is and simply remove these files from templates.yaml, or copy-paste the one from reuseble-workflows.

If we try to adjust the workflows that exist here, we might face some issues, because there is no easy way to test them

**Comment by dpotyralski** [comment]
> @dpotyralski Is this a copy-paste from the new cicd workflow, or you manipulated the files that exist here?
> 
> I think there are 2 "safe" options. Either keep everything as it is and simply remove these files from templates.yaml, or copy-paste the one from reuseble-workflows.
> 
> If we try to adjust the workflows that exist here, we might face some issues, because there is no easy way to test them

Those existing ones were distributed here from our templates, so are having the latest changes. I've decided to additionally clean them up from functionalities, which are not longer relevant.

**Review by DimitrisKaramanis — APPROVED** [review]

**Comment by dpotyralski** [comment]
Hey @pleo-io/team-invoices, would appreciate your eyes on this one 🙏

**Review by blade47 — APPROVED** [review]

**Pull request merged**